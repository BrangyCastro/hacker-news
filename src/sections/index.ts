export { Alls } from "./Alls/Alls";
export { Favorite } from "./Favorite/Favorite";
