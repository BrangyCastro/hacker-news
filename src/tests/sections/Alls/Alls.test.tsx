import React from "react";
import { shallow } from "enzyme";
import { Alls } from "../../../sections";
import { dataNews } from "../../fixtures/demoNews";

describe("Test section alls news", () => {
  test("should display correctly without news", () => {
    const myInitialState = {
      news: [],
      page: 0,
      loadingFilter: true,
    };
    React.useState = jest.fn().mockReturnValue([myInitialState, {}]);
    const wrapper = shallow(<Alls />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h1").text().trim()).toBe("Loading news...");
  });

  test("should display correctly with news", () => {
    const myInitialState = {
      news: dataNews,
      page: 0,
      loadingFilter: false,
    };
    React.useState = jest.fn().mockReturnValue([myInitialState, {}]);
    const wrapper = shallow(<Alls />);
    expect(wrapper.find("Card").exists()).toBe(true);
  });

  test("should be three news", () => {
    const myInitialState = {
      news: dataNews,
      page: 0,
      loadingFilter: false,
    };
    React.useState = jest.fn().mockReturnValue([myInitialState, {}]);
    const wrapper = shallow(<Alls />);
    const card = wrapper.find("Card");
    expect(card).toHaveLength(dataNews.length);
  });
});
