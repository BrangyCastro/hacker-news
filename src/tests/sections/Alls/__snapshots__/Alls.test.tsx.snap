// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test section alls news should display correctly without news 1`] = `
<Fragment>
  <Dropdown
    onChange={[Function]}
    options={
      Array [
        Object {
          "icon": "angular.png",
          "label": "Angular",
          "value": "Angular",
        },
        Object {
          "icon": "reactjs.png",
          "label": "Reactjs",
          "value": "Reactjs",
        },
        Object {
          "icon": "vuejs.png",
          "label": "Vuejs",
          "value": "Vuejs",
        },
      ]
    }
    placeholder="Select your news"
  />
  <h1
    className="loading"
  >
    Loading news...
  </h1>
</Fragment>
`;
